model/src/main/java/com/redhat/ipaas/model/integration/Integration.java

```diff
@@ -65,6 +65,8 @@ default Kind getKind() {
 
     Optional<Status> getCurrentStatus();
 
+    Optional<List<String>> getStepsDone();
```

Comment:
I'm not keen on adding steps done to the integration, as this will be sent to the clients too, which I guess isn't so bad in hindsight. Once we review the API models though, I'd prefer to differentiate between spec (desired) and status (current) as is done in Kubernetes API. That simplifies things quite a lot, easier to read, and not conflate current and requested state. Don't worry about changing this now though. Full review post-summit.